wrappers/services.go

```diff
@@ -582,9 +582,42 @@ func (es *ensureSnapServicesContext) reloadModified() error {
 	return nil
 }
 
+// quotaGroupByName is a helper to find the quota group by name in a quota group slice.
+func quotaGroupByName(name string, allGrps []*quota.Group) *quota.Group {
+	for _, grp := range allGrps {
+		if grp.Name == name {
+			return grp
+		}
+	}
+	return nil
+}
+
+// resolveAppSpecificQuotaGroup takes into account any service-specific quota group. Quota groups
+// are assigned on snap basis in general, which means the quota applies to everything inside the snap, unless
+// a sub-group exists that refer to a specific service. This function checks for that, and returns the correct
+// quota group, should a sub-group exist for this specific appInfo.
+func resolveAppSpecificQuotaGroup(appInfo *snap.AppInfo, snapQuotaGroup *quota.Group, allGrps []*quota.Group) *quota.Group {
+	if snapQuotaGroup == nil {
+		return nil
+	}
+
+	for _, name := range snapQuotaGroup.SubGroups {
+		subGroup := quotaGroupByName(name, allGrps)
```

Comment:
if we need to iterate over all the groups anyway I would do something else, intestead of going over the subgroups, I would go over all the groups check in the parent is snapQuotaGroup and then check if the app is in this group then. This a makes us iterate over all the groups only once